scripts/common_functions.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+# Implements a library of commonly used functions, namely:
+# - echo_action
+# - echo_error
+# - echo_info
+# - echo_title
+# - parse_parameters
+# - set_trap
+# - unset_trap
+
+# Constants
+readonly DATE_FORMAT='%Y-%m-%d %H:%M:%S (%Z)'
+
+# Functions
+function echo_action() {
+  echo ""
+  echo "$(date +"$DATE_FORMAT") | ACTION - $1"
+}
+
+function echo_error() {
+  echo "$(date +"$DATE_FORMAT") | ERROR  - $1" >&2
+}
+
+function echo_info() {
+  echo "$(date +"$DATE_FORMAT") | INFO   - $1"
+}
+
+function echo_title() {
+  echo ""
+  echo "###############################################################################"
+  echo "$1"
+  echo "###############################################################################"
+}
+
+# Assumptions:
+# - A global associative array named "parameters" exists.
+function parse_parameters() {
+  local readonly PARAMETER_KEY_PREFIX="--"
+  local readonly PARAMETER_KEY_REGEX_PATTERN="^${PARAMETER_KEY_PREFIX}.*$"
+
+  local missing_parameter_flag=false
+  local sorted_parameter_keys=$(echo ${!parameters[@]} | tr " " "\n" | sort | tr "\n" " ");
+  local unexpected_parameter_flag=false
+  local usage="USAGE: $(basename $0)"
+
+  echo_action "Mapping input parameter values and checking for unexpected parameters..."
+  while [[ ${#@} -gt 0 ]];
```

Comment:
Put ; do and ; then on the same line as the while, for or if.

Reply:
Effectivement.